packages/react-native-bpk-component-button/src/BpkButton.android.js

```diff
@@ -19,108 +19,223 @@
+    return {
+      backgroundColor: themeAttributes
+        ? themeAttributes.buttonFeaturedGradientStartColor
+        : backgroundColors.featured,
+    };
+  }
+  if (type === BUTTON_TYPES.secondary) {
+    if (disabled) {
+      return {
+        backgroundColor: backgroundColors.disabled,
+        borderColor: borderColors.disabled,
+      };
+    }
+    return themeAttributes
+      ? {
+          backgroundColor: themeAttributes.buttonSecondaryBackgroundColor,
+          borderColor: themeAttributes.buttonSecondaryBorderColor,
+        }
+      : {
+          backgroundColor: backgroundColors.secondary,
+          borderColor: borderColors.secondary,
+        };
+  }
+  if (disabled) {
+    return {
+      backgroundColor: backgroundColors.disabled,
+      borderColor: borderColors.disabled,
+    };
+  }
+  return themeAttributes
+    ? {
+        backgroundColor: themeAttributes.buttonDestructiveBackgroundColor,
+        borderColor: themeAttributes.buttonDestructiveBorderColor,
+      }
+    : {
+        backgroundColor: backgroundColors.destructive,
+        borderColor: borderColors.destructive,
+      };
+};
+
+const getTextColor = (
+  type: ButtonType,
+  themeAttributes: ?Theme,
+  disabled: boolean,
+) => {
+  if (disabled) {
+    return textColors.disabled;
+  }
+  switch (type) {
+    case BUTTON_TYPES.primary:
+      return themeAttributes
+        ? themeAttributes.buttonPrimaryTextColor
+        : textColors.primary;
+    case BUTTON_TYPES.featured:
+      return themeAttributes
+        ? themeAttributes.buttonFeaturedTextColor
+        : textColors.featured;
+    case BUTTON_TYPES.secondary:
+      return themeAttributes
+        ? themeAttributes.buttonSecondaryTextColor
+        : textColors.secondary;
+    default:
+      return themeAttributes
+        ? themeAttributes.buttonDestructiveTextColor
+        : textColors.destructive;
+  }
 };
 
 const BpkButton = (props: Props) => {
   const {
     accessibilityLabel,
     disabled,
     icon,
+    iconAlignment,
     iconOnly,
-    onPress,
-    style,
+    theme,
     title,
     type,
-    theme: themeProp,
     ...rest
   } = props;
 
-  if (!Object.values(BUTTON_TYPES).includes(type)) {
+  if (!Object.keys(BUTTON_TYPES).includes(type)) {
     throw new Error(
       `"${type}" is not a valid button type. Valid types are ${Object.keys(
         BUTTON_TYPES,
       ).join(', ')}`,
     );
   }
 
-  const shouldApplyTheme =
-    themeProp &&
-    (isTypeThemeable(type) && themeAttributesSupplied(type, themeProp));
-  const theme = shouldApplyTheme ? themeProp : null;
+  const themeAttributes = getThemeAttributes(
+    REQUIRED_THEME_ATTRIBUTES[type],
+    theme,
+  );
 
-  const accessibilityTraits = ['button'];
-  const containerStyle = getStyleForElement('androidContainer', props);
-  const buttonStyle = getStyleForElement('button', props);
-  const backgroundColor = getAndroidBackgroundColour(theme, props);
+  const ButtonComponent = getButtonComponent(type);
```

Comment:
nitpic: I am personally not the biggest fan of the `get` prefix. Maybe it's my Swift/Obejctive-C experience, but `buttonComponentForType(type)` feels much nicer

Reply:
That's fine - the naming was very arbitrary and your way sounds nicer.